src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStaticTests.cs

```diff
@@ -577,6 +577,70 @@ End Class
             }.RunAsync();
         }
 
+        [Fact, WorkItem(3019, "https://github.com/dotnet/roslyn-analyzers/issues/3019")]
+        public async Task PrivateMethodOnlyCalledByASkippedMethod_Diagnostic()
+        {
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                        @"
+using Xunit;
+
+public class Program
+{
+    [Fact]
```

Comment:
I only made a test with `Fact` as it is the one reported in the ticket. Is it worth adding test cases for all of the exit conditions of the `ShouldAnalyze` method?

Reply:
I think this coverage is fine.